### DISCUSSION TITLE
Help learning how to code

### DISCUSSION THREAD

**Discussion by Leptik2wo** [TheAlgorithms/Python#11505] [Q&A]
Ive never done anything with coding at all but really want to learn. Does anyone have any pointers of where I should start or how I should do it?

**Comment by yash-saini** [comment]
I believe you should start by any language, Java/Python/C#/C++. Search for tutorials on youtube, then practise problems on leetcode / Hackerrank.

**Comment by imSanko** [comment]
@Leptik2wo 
1. Online Courses:
    - Codecademy (interactive coding lessons)
    - Coursera (university-level courses)
    - Udemy (affordable courses)
2. Tutorials and Guides:
    - FreeCodeCamp (web development)
    - W3Schools (web development)
    - GeeksforGeeks (programming concepts)



**-----------Mark it as answered if it helped-----------**

**Reply by jotavejotave** [reply]
> @Leptik2wo
> 
> 1. Online Courses:
>    
>    * Codecademy (interactive coding lessons)
>    * Coursera (university-level courses)
>    * Udemy (affordable courses)
> 2. Tutorials and Guides:
>    
>    * FreeCodeCamp (web development)
>    * W3Schools (web development)
>    * GeeksforGeeks (programming concepts)
> 
> **-----------Mark it as answered if it helped-----------**

thanks

**Comment by Temegw21** [comment]
The first step is to identify your career goal: Are you aiming to be a mobile app developer, a full-stack developer, a data scientist, or something else? Once you have a clear direction, choose a programming language that aligns with your aspirations. Remember, regardless of the language you select, the foundational programming concepts you learn will remain consistent; the main differences will be in syntax.

**Comment by HumaidhMilhan** [comment]
The answer to this question lies in your goals. First figure what's the purpose that you are learning for. Then research what's the language that suits best for that purpose. Then start from the basics. The starting point might be confusing if you're new to coding. Then gradually increase the pace as you start to understand. 

So, the first part of the answer to this question is the question of "What's the purpose of your learning?". Then research accordingly for that purpose. 

I started with no idea at the beginning of the year and was going hard on myself trying to learn multiple languages and without a goal of what I'm learning for. Everything was a mess that way. But when you figure out the purpose, it makes everything a lot easier. 

This answer is from my experience and I think that this suits all beginners who are starting to code. Have a great day!

**Comment by dryasirmirza** [comment]
Dear, first of all clear your mind that whether you are willing to learn or not. If you are dedicated to learn then there are various open source videos on YouTube. Just start with anyone, anyone. Don't look for something best and best, just click on the video and start. If you are able to learn dedicatedly atleast for two hours then nothing can stop you from learning. All doors will be open for you. 

Just make your mind clear to learn.
